packages/main/src/dynamic-date-range-options/LastOptions.ts

```diff
@@ -0,0 +1,173 @@
+import LastNextTemplate from "./LastNextTemplate.js";
+import type { DynamicDateRangeValue, IDynamicDateRangeOption } from "../DynamicDateRange.js";
+import type { JsxTemplate } from "@ui5/webcomponents-base/dist/index.js";
+import type { I18nText } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import {
+	formatLastNext, handleSelectionChangeLastNext, isValidStringLastNext, toDatesLastNext, parseLastNext,
+} from "./LastNextUtils.js";
+import DynamicDateRange from "../DynamicDateRange.js";
+import {
+	DYNAMIC_DATE_RANGE_LAST_DAYS_TEXT,
+	DYNAMIC_DATE_RANGE_LAST_WEEKS_TEXT,
+	DYNAMIC_DATE_RANGE_LAST_MONTHS_TEXT,
+	DYNAMIC_DATE_RANGE_LAST_QUARTERS_TEXT,
+	DYNAMIC_DATE_RANGE_LAST_YEARS_TEXT,
+} from "../generated/i18n/i18n-defaults.js";
+
+class LastOptions implements IDynamicDateRangeOption {
+	template: JsxTemplate = LastNextTemplate;
+	operator: string;
+	i18nKey: I18nText;
+
+	constructor(operator: string, i18nKey: I18nText) {
+		this.operator = operator;
+		this.i18nKey = i18nKey;
+	}
+
+	parse(value: string): DynamicDateRangeValue | undefined {
+		const matchingOption = this.availableOptions.find(optionInfo => {
+			const tempOption = { operator: optionInfo.operator, text: optionInfo.text };
+			return isValidStringLastNext(value, tempOption as IDynamicDateRangeOption);
+		});
+
+		if (matchingOption) {
+			const tempOption = { operator: matchingOption.operator, text: matchingOption.text };
+			return parseLastNext(value, tempOption as IDynamicDateRangeOption);
+		}
+
+		return undefined;
+	}
+
+	format(value: DynamicDateRangeValue): string {
+		const optionInfo = this.availableOptions.find(info => info.operator === value.operator);
+		if (optionInfo) {
+			const numberValue = this._getNumberFromValue(value);
+			const tempValue = { operator: value.operator, values: [numberValue] };
+			return formatLastNext(tempValue, { text: optionInfo.text } as IDynamicDateRangeOption);
+		}
+		return "";
+	}
+
+	toDates(value: DynamicDateRangeValue): Array<Date> {
+		return toDatesLastNext(value, { operator: value.operator } as IDynamicDateRangeOption);
+	}
+
+	isValidString(value: string): boolean {
+		return this.availableOptions.some(optionInfo => {
+			const tempOption = { operator: optionInfo.operator, text: optionInfo.text };
+			return isValidStringLastNext(value, tempOption as IDynamicDateRangeOption);
+		});
+	}
+
+	handleSelectionChange(e: CustomEvent): DynamicDateRangeValue | undefined {
+		return handleSelectionChangeLastNext(e, this);
+	}
+
+	get text(): string {
+		const currentOptions = DynamicDateRange.getCurrentOptions();
+		if (currentOptions) {
+			const lastOptions = currentOptions.split(",")
+				.map(s => s.trim())
+				.filter(s => s.startsWith("LAST"));
+
+			if (lastOptions.length > 1) {
+				const units = this.availableOptions
+					.filter(info => lastOptions.includes(info.operator))
+					.map(info => info.unitText);
+				return `Last X ${units.join(" / ")} (included)`;
```

Comment:
did we decide to have included or excluded?

Reply:
Yes, to my latest knowledge we decided to go with included.